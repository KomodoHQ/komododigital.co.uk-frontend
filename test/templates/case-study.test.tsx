import React from 'react';
import CaseStudy from '../../src/templates/case-study';
import renderer from 'react-test-renderer';
import GatsbyData from '../../gatsby-config';

test('Case Study template is rendered', () => {

  const htmlAst = {
    type: 'element',
    tagName: 'span',
    properties: {},
    children: [
      {
        type: 'text',
        value: 'Lorem Ipsum is the industry standard typesetting text',
      },
    ],
  };

  const hocProps = {
    caseStudies: [
      {
        htmlAst: htmlAst,
        link: "case-studies/start-up",
        coverimage: {
          childImageSharp: {
            fluid: {
              height: 29,
              src: "",
              srcSet: "",
              srcSetWebp: "",
              srcWebp: "",
              tracedSVG: "",
              width: 170,
              aspectRatio: 1,
              sizes: ""
            }
          }
        },
        csimage: {
          childImageSharp: {
            fluid: {
              height: 29,
              src: "",
              srcSet: "",
              srcSetWebp: "",
              srcWebp: "",
              tracedSVG: "",
              width: 170,
              aspectRatio: 1,
              sizes: ""
            }
          }
        },
        key: "study.title",
        subtitle:"study.subtitle",
        title:"study.title"
      }
    ],
    metrics: [],
    subtitle: 'subtitle',
    title: 'title',
    intro: htmlAst,
    metricsIntro: htmlAst,
    processTitle: 'title',
    process: htmlAst,
    caseStudiesIntro: htmlAst,
    contactsIntro: htmlAst,
    data: {
      site: GatsbyData,
      logo: {
        childImageSharp: {
          fixed: {
            height: 29,
            src: "",
            srcSet: "",
            srcSetWebp: "",
            srcWebp: "",
            tracedSVG: "",
            width: 170
          }
        }
      },
      logo_inverted: {
        childImageSharp: {
          fixed: {
            height: 29,
            src: "",
            srcSet: "",
            srcSetWebp: "",
            srcWebp: "",
            tracedSVG: "",
            width: 170
          }
        }
      }

    },
  };

  const component = renderer.create(<CaseStudy {...hocProps} />);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
