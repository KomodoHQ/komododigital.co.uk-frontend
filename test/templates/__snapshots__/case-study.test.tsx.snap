// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Case Study template is rendered 1`] = `
Array [
  <noscript>
    You have Javascript disabled. While it isn't needed, it will make your experience nicer.
  </noscript>,
  <div
    className="komodoGridWrapper navigationBar "
    style={
      Object {
        "background": "#12203A",
      }
    }
  >
    <a
      aria-current="page"
      href="/"
      onClick={[Function]}
    >
      <div
        className=" gatsby-image-wrapper"
        style={
          Object {
            "display": "inline-block",
            "height": 29,
            "overflow": "hidden",
            "position": "relative",
            "width": 170,
          }
        }
      >
        
        <picture>
          
          <source
            srcSet=""
          />
          <img
            alt=""
            height={29}
            onLoad={[Function]}
            src=""
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 1,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
            width={170}
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><img width=\\"170\\" height=\\"29\\" src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </a>
    <div
      className="hamburger"
    >
      <a
        aria-current="page"
        href="/"
        onClick={[Function]}
      >
        <img
          height={64}
          src={Object {}}
          style={
            Object {
              "height": 64,
              "marginTop": 10,
              "width": 64,
            }
          }
          width={64}
        />
      </a>
    </div>
    <ul
      className="Menu"
      style={
        Object {
          "background": "#12203A",
        }
      }
    >
      <li>
        <a
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
      <li>
        <a
          href="/case-studies"
          onClick={[Function]}
        >
          Case Studies
        </a>
      </li>
      <li>
        <a
          href="/blog-list"
          onClick={[Function]}
        >
          Insights
        </a>
      </li>
      <li>
        <a
          href="/contact"
          onClick={[Function]}
        >
          Contact
        </a>
      </li>
    </ul>
  </div>,
  <div
    className="main"
    style={
      Object {
        "background": "linear-gradient(to bottom, #12203A 80px,#EAEAEA 80px)",
      }
    }
  >
    <div
      className="komodoGridWrapper cs-wrapper invert topPaddingLarge bottomPaddingLarge"
      style={
        Object {
          "background": "linear-gradient(0deg, #23375A 0%, #12203A 100%)",
        }
      }
    >
      <div
        className="Content-Section"
      >
        <span>
          subtitle
        </span>
        <h2>
          title
        </h2>
        <div>
          <span>
            Lorem Ipsum is the industry standard typesetting text
          </span>
        </div>
      </div>
      <div
        className=" gatsby-image-wrapper"
        style={
          Object {
            "overflow": "hidden",
            "position": "relative",
          }
        }
      >
        <div
          style={
            Object {
              "paddingBottom": "100%",
              "width": "100%",
            }
          }
        />
        
        <picture>
          
          <source
            sizes=""
            srcSet=""
          />
          <img
            alt=""
            onLoad={[Function]}
            src=""
            style={
              Object {
                "height": "100%",
                "left": 0,
                "objectFit": "cover",
                "objectPosition": "center",
                "opacity": 1,
                "position": "absolute",
                "top": 0,
                "transition": "opacity 0.5s",
                "width": "100%",
              }
            }
          />
        </picture>
        <noscript
          dangerouslySetInnerHTML={
            Object {
              "__html": "<picture><img width=\\"170\\" height=\\"29\\" src=\\"\\" alt=\\"\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
            }
          }
        />
      </div>
    </div>
    <div
      className="komodoGridWrapper cs-wrapper  "
      style={
        Object {
          "background": "none",
        }
      }
    >
      <div
        className="Content-Section"
      >
        <h2>
          Solution/Project Outcome
        </h2>
        <div>
          <span>
            Lorem Ipsum is the industry standard typesetting text
          </span>
        </div>
      </div>
    </div>
    <div
      className="komodoGridWrapper metrics-wrapper"
    >
      <div
        className="Metrics"
      />
    </div>
    <div
      className="komodoGridWrapper cs-wrapper  "
      style={
        Object {
          "background": "none",
        }
      }
    >
      <div
        className="Content-Section"
      >
        <h2>
          title
        </h2>
        <div>
          <span>
            Lorem Ipsum is the industry standard typesetting text
          </span>
        </div>
      </div>
    </div>
    <article
      className="komodoGridWrapper testimonial-wrapper"
      style={
        Object {
          "background": "linear-gradient(0deg, #23375A 0%, #12203A 100%)",
        }
      }
    >
      <div
        className="Testimonial"
      >
        <span
          className="name"
        >
          Paul Crennell
          :
        </span>
        <q>
          Komodo are very refreshing, a talented, engaging and flexible team completely focused on delivering the customer's objectives.
        </q>
        <hr />
        <span
          className="job"
        >
          Chief Technology Officer
        </span>
        <span
          className="company"
        >
          Orchard Systems
        </span>
      </div>
    </article>
    <div
      className="komodoGridWrapper cs-wrapper  "
      style={
        Object {
          "background": "none",
        }
      }
    >
      <div
        className="Content-Section"
      >
        <h2>
          Other related work
        </h2>
        <div>
          <span>
            Lorem Ipsum is the industry standard typesetting text
          </span>
        </div>
      </div>
    </div>
    <div
      className="komodoGridWrapper case-study-link-wrapper"
    >
      <div
        className="case-study-link"
      >
        <div
          className="img"
        >
          <div
            className=" gatsby-image-wrapper"
            style={
              Object {
                "overflow": "hidden",
                "position": "relative",
              }
            }
          >
            <div
              style={
                Object {
                  "paddingBottom": "100%",
                  "width": "100%",
                }
              }
            />
            
            <picture>
              
              <source
                sizes=""
                srcSet=""
              />
              <img
                alt=""
                onLoad={[Function]}
                src=""
                style={
                  Object {
                    "height": "100%",
                    "left": 0,
                    "objectFit": "cover",
                    "objectPosition": "center",
                    "opacity": 1,
                    "position": "absolute",
                    "top": 0,
                    "transition": "opacity 0.5s",
                    "width": "100%",
                  }
                }
                title="study.title"
              />
            </picture>
            <noscript
              dangerouslySetInnerHTML={
                Object {
                  "__html": "<picture><img width=\\"170\\" height=\\"29\\" src=\\"\\" alt=\\"\\" title=\\"study.title\\" style=\\"position:absolute;top:0;left:0;transition:opacity 0.5s;transition-delay:0.5s;opacity:1;width:100%;height:100%;object-fit:cover;object-position:center\\"/></picture>",
                }
              }
            />
          </div>
        </div>
        <div
          className="content"
        >
          <span>
            study.subtitle
          </span>
          <h2>
            study.title
          </h2>
          <span>
            Lorem Ipsum is the industry standard typesetting text
          </span>
          <a
            href="/case-studies/start-up"
            onClick={[Function]}
          >
            Read Case Study
          </a>
        </div>
      </div>
    </div>
    <div
      className="komodoGridWrapper cs-wrapper  "
      style={
        Object {
          "background": "#ffffff",
        }
      }
    >
      <div
        className="Content-Section"
      >
        <h2>
          Contact Us
        </h2>
        <div>
          <span>
            Lorem Ipsum is the industry standard typesetting text
          </span>
        </div>
      </div>
    </div>
    <div
      className="komodoGridWrapper contactsection-wrapper"
    >
      <div
        className="contactsection"
      >
        <div
          className="contactform"
        >
          <form
            action="https://formspree.io/hello@komododigital.co.uk"
            method="POST"
          >
            <label
              htmlFor="form-name"
            >
              Your Name
              <input
                id="form-name"
                name="name"
                placeholder="Joe Blogs"
                type="text"
              />
            </label>
            <label
              htmlFor="form-email"
            >
              Your Email
              <input
                id="form-email"
                name="email"
                placeholder="joe@blogs.com"
                type="email"
              />
            </label>
            <label
              htmlFor="form-service"
            >
              Service
              <select
                id="form-service"
                name="service"
              >
                <option>
                  Choose Service
                </option>
                <option
                  value="Discovery"
                >
                  Discovery
                </option>
                <option
                  value="Design"
                >
                  Design
                </option>
                <option
                  value="Development"
                >
                  Development
                </option>
                <option
                  value="Delivery"
                >
                  Delivery
                </option>
              </select>
            </label>
            <label
              htmlFor="form-sector"
            >
              Sector
              <select
                id="form-sector"
                name="sector"
              >
                <option>
                  Choose Sector
                </option>
                <option
                  value="Education"
                >
                  Education
                </option>
                <option
                  value="Law Enforcement"
                >
                  Law Enforcement
                </option>
                <option
                  value="Startups"
                >
                  Startups
                </option>
                <option
                  value="Social Housing"
                >
                  Social Housing
                </option>
              </select>
            </label>
            <label
              htmlFor="form-message"
            >
              Your Message
              <textarea
                cols={30}
                id="form-message"
                name="message"
                placeholder="Your message"
                rows={10}
              />
            </label>
            <div>
              <input
                type="submit"
                value="Send"
              />
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>,
  <footer
    className="komodoGridWrapper"
  >
    <div
      className="komodo"
    >
      <img
        alt="Komodo - People Friendly Software"
        src={Object {}}
      />
    </div>
    <div
      className="address"
    >
      <span>
        Address
      </span>
      <address>
        Komodo Digital,
        <br />
        63 Westgate Rd,
        <br />
        Newcastle upon Tyne,
        <br />
        NE1 1SG
      </address>
    </div>
    <div
      className="sitemap"
    >
      <span>
        Sitemap
      </span>
      <ul>
        <li>
          <a
            aria-current="page"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            href="/"
            onClick={[Function]}
          >
            Case Studies
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            href="/"
            onClick={[Function]}
          >
            Insights
          </a>
        </li>
        <li>
          <a
            aria-current="page"
            href="/"
            onClick={[Function]}
          >
            Careers
          </a>
        </li>
      </ul>
      <ul>
        <li>
          <a
            aria-current="page"
            href="/"
            onClick={[Function]}
          >
            Contact
          </a>
        </li>
        <li>
          <a
            href="https://twitter.com/komodohq"
          >
            Twitter
          </a>
        </li>
        <li>
          <a
            href="https://facebook.com/komodohq"
          >
            Facebook
          </a>
        </li>
        <li>
          <a
            href="https://linkedin.com/komodohq"
          >
            LinkedIn
          </a>
        </li>
      </ul>
    </div>
  </footer>,
  <footer
    className="komodoGridWrapper legal-footer"
  >
    <div
      className="company"
    >
      <span>
        Komodo Digital Copyright © 2018
      </span>
      <br />
      <a
        aria-current="page"
        href="/"
        onClick={[Function]}
      >
        Privacy Policy
      </a>
       | 
      <a
        aria-current="page"
        href="/"
        onClick={[Function]}
      >
        Terms and Conditions
      </a>
    </div>
    <div
      className="contact"
    >
      Get in touch on:
      <br />
      <a
        href="/tel:0191 228 6555"
        onClick={[Function]}
      >
        0191 228 6555
      </a>
    </div>
  </footer>,
  <noscript>
    <img
      alt=""
      height="1"
      src="https://dc.ads.linkedin.com/collect/?pid=377068&fmt=gif"
      width="1"
    />
  </noscript>,
]
`;
